components/scream/src/physics/shoc/shoc_functions_f90.hpp

```diff
@@ -323,6 +323,84 @@ struct SHOCMixcheckData : public PhysicsTestData {
   PTD_ASSIGN_OP(SHOCMixcheckData, 0);
 };//SHOCMixcheckData
 
+//Create data structure to hold data for clipping_diag_third_shoc_moments
+struct SHOCClipthirdmomsData : public PhysicsTestData {
+  // Inputs
+  Real *w_sec_zi;
+
+  // In/out
+  Real *w3;
+
+  SHOCClipthirdmomsData(Int shcol_, Int nlev_, Int nlevi_) :
+    PhysicsTestData(shcol_, nlev_, nlevi_,{},{&w_sec_zi, &w3}){}
```

Comment:
This doesn't look right. You just want shcol of w_sec and w3, right? Then you just need:
```
PhysicsTestData(shcol_, {&w_sec_zi, &w3}) {}
```

Reply:
those arrays are actually of shape (shcol,nlevi).  I'm struggling how to do this correctly when there is data of (shcol,nlevi) but not data of (shcol,nlev).  The way I thought it should be done is:

SHOCClipthirdmomsData(Int shcol_, Int nlevi_):
  PhysicsTestData(shcol_, nlevi_, {&w_sec_zi, &w3}){}

and changing the 3 in PTD_DATA_COPY_CTOR to 2.  This compiles, but the test fails because nlevi is never assigned a value.  The only way I could get it to work was to assign a value for nlev (even though it's never used) and use the empty array.  Clearly I'm thinking about this wrong...

Reply:
@bogensch , I see. You are not thinking about it wrong... I think I need to rethink the naming in PhysicsTestData.

Reply:
@bogensch , once #570 is in, you can name the dimensions whatever you want.

Reply:
Thanks, i've merged now with master (I've pushed my changes if you want to look), but I'm still having trouble which is probably my fault.  Just to clarify should the correct syntax for this block now be:

  SHOCClipthirdmomsData(Int shcol_, Int nlevi_) :
    PhysicsTestData(shcol_, nlevi_,{&w_sec_zi, &w3}){}

  SHOC_NO_SCALAR(SHOCClipthirdmomsData, 2);
 
?  If so it complains that there is no d.nlevi() in the shoc_functions_f90.cpp file in:

void clipping_diag_third_shoc_moments(SHOCClipthirdmomsData &d) {
  shoc_init(d.nlev(), true);
  d.transpose<ekat::util::TransposeDirection::c2f>();
  clipping_diag_third_shoc_moments_c(d.nlevi(),d.shcol(),d.w_sec_zi,d.w3);
  d.transpose<ekat::util::TransposeDirection::f2c>();
}

If I change d.nlevi() to d.nlev(), then I can get the test to compile/pass.  Does this seem right to you?

Reply:
@bogensch , that makes sense. The SHOC_NO_SCALAR macro automatically renames the first dimension shcol, the second nlev, and the third nlevi since this is the common pattern in SHOC*Data structs. If you want a different naming scheme, put this in your struct instread of SHOC_NO_SCALAR:
```
  PTD_STD_DEF(SHOCClipthirdmomsData, 2, 0);
  PTD_DIM_RENAME(2, shcol, nlevi)
```

Reply:
thanks!